### PR TITLE
Add "all-public-access" optional component for automated integration testing

### PR DESCRIPTION
**Pull request by MatProv**
When enabled, allows authtest user and anonymous group to access all wps endpoints and thredds.

Note that the authtest user needs to be created with the create-magpie-user script.

### CODE REVIEW

File: birdhouse/optional-components/all-public-access/all-public-access-magpie-permission.cfg [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    user: authtest
+    action: create
+
+  - service: hummingbird
+    permission: execute
+    user: authtest
+    action: create
+
+
+  - service: lb_flyingpigeon
```
Comment (tlvu) [inline birdhouse/optional-components/all-public-access/all-public-access-magpie-permission.cfg:66]: I would remove all mentions of lb_flyingpigeon, no tests are targetting it.  It is an extra config that is not in this repo.  In fact it's probably a mistake to have left it in `birdhouse/config/magpie/providers.cfg.template`, sorry.  I'll fix this in a different PR I need to sync this with our "private config" that will restore it.

File: birdhouse/optional-components/all-public-access/all-public-access-magpie-permission.cfg [hunk 2]
```diff
@@ -0,0 +1,235 @@
+permissions:
+  - service: catalog
+    permission: getcapabilities
+    user: authtest
```
Comment (tlvu) [inline birdhouse/optional-components/all-public-access/all-public-access-magpie-permission.cfg:4]: You don't have to repeat the exact same authorizations as group `anonymous` for user `authtest`, he's part of that group anyways so it's redundant.  

However you need to add read permissions for folder https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/catalog/birdhouse/testdata/secure/catalog.html that group `anonymous` should not have.

I know the current version of Magpie do not support removing read permissions if  the parent already has it.  The next Magpie should be able to do this.

**Review by tlvu — CHANGES_REQUESTED** [review]
LGTM, minor changes requested.

Please also add something to the https://github.com/bird-house/birdhouse-deploy/blob/master/birdhouse/optional-components/README.md mentioning this component will open up public access to all services listed in `birdhouse/config/magpie/providers.cfg.template`, user beware when enable this, that this is useful for the test suite in https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests.  Follow existing format description of other optional-components.

The `.gitignore` file is empty, please remove it.

**Comment by tlvu** [comment]
@MatProv will work in another PR, I'll include the `bootstrap-testdata` script you've used, a `trigger-crawler` script, sample config to create that `authtest` user and update the README to mentions "steps" to make the new instance "test ready".  Anything I forgot to get the fresh instance ready for our Jenkins test suite?

File: birdhouse/optional-components/all-public-access/all-public-access-magpie-permission.cfg (hunk 2 above)
Comment (MatProv) [reply birdhouse/optional-components/all-public-access/all-public-access-magpie-permission.cfg:4]: The `group:anonymous` permissions are removed, so it won't be redundant.
And for the read access to thredds, it's included in the permissions (see latest block).

```
 - service: thredds
    permission: read
    user: authtest
    action: create
```

**Comment by MatProv** [comment]
> @MatProv will work in another PR, I'll include the `bootstrap-testdata` script you've used, a `trigger-crawler` script, sample config to create that `authtest` user and update the README to mentions "steps" to make the new instance "test ready". Anything I forgot to get the fresh instance ready for our Jenkins test suite?

Since I've removed rights for anonymous data access, notebooks accessing data anonymously would then need to be fixed.

Other than that changes you propose seems fine to me.

File: birdhouse/optional-components/all-public-access/all-public-access-magpie-permission.cfg (hunk 2 above)
Comment (tlvu) [reply birdhouse/optional-components/all-public-access/all-public-access-magpie-permission.cfg:4]: oh no, you should keep group `anonymous` and remove user `authtest`, except when you give it extra perm to read the `secure` folder.

All the testsuite use anonymous access !

Comment (MatProv) [reply birdhouse/optional-components/all-public-access/all-public-access-magpie-permission.cfg:4]: Oh alright, so i'll set all the wps permissions for anonymous, and keep the latest block about authtest and thredds.

File: birdhouse/optional-components/all-public-access/all-public-access-magpie-permission.cfg [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  - service: thredds
+    permission: read
+    group: anonymous
+    action: create
+
+
+  - service: thredds
+    permission: read
+    user: authtest
```
Comment (tlvu) [inline birdhouse/optional-components/all-public-access/all-public-access-magpie-permission.cfg:106]: This is same as for group anonymous above, why need it? 

If to give specific read access to the secure folder, I think you'll have to add `resource: /birdhouse/testdata/secure` (untested syntax).

File: birdhouse/optional-components/README.md [hunk 4]
```diff
@@ -112,3 +112,26 @@ Magpie will be automatically configured to give complete public anonymous
 access for this WPS service.
 
 Canarie monitoring will also be automatically configured for this WPS service.
+
+
+## All access given to test user
```
Comment (tlvu) [inline birdhouse/optional-components/README.md:117]: Suggested text change: "Give public access to all resources for testing purposes"

File: birdhouse/optional-components/README.md [hunk 5]
```diff
@@ -112,3 +112,26 @@ Magpie will be automatically configured to give complete public anonymous
...
+
+Can be used for testing purposes.
```
Comment (tlvu) [inline birdhouse/optional-components/README.md:119]: suggested text change "By enabling this component, all WPS services and data on Thredds are completely public, please beware.  Once enabled, if you need to revert the change, you have to do it manually by logging into Magpie.  Just disabling this component will not revert the change."

File: birdhouse/optional-components/README.md [hunk 6]
```diff
@@ -112,3 +112,26 @@ Magpie will be automatically configured to give complete public anonymous
...
+Allows the tests in https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests to pass with the `authtest/authtest` user.
```
Comment (tlvu) [inline birdhouse/optional-components/README.md:120]: Remove all mentions of the `authtest` user and instructions to create it, I've create a script to do it in my other PR.  For security reasons, let's not broadcast too loudly this hardcoded user with known hardcoded password.

**Review by tlvu — CHANGES_REQUESTED** [review]
Much better, minor changes requested.  By the way, lost the http(S) on the link to http://daccs-jenkins.crim.ca again, did you forget to commit jenkins config change to jenkins?

**Review by tlvu — APPROVED** [review]
+1

**Comment by MatProv** [comment]
> By the way, lost the http(S) on the link to http://daccs-jenkins.crim.ca again, did you forget to commit jenkins config change to jenkins?

I've simply put this aside for the moment, was not a priority as for the tests that need to pass.
We will address this later on.
